pkg/storage/ingest/fetcher.go

```diff
@@ -466,14 +465,14 @@ func (g offsetRange) numOffsets() int64 {
 	return g.end - g.start
 }
 
-func findGapsInRecords(records []*kgo.Record, lastReturnedOffset int64) []offsetRange {
+func findGapsInRecords(records kgo.Fetches, lastReturnedOffset int64) []offsetRange {
 	var gaps []offsetRange
```

Comment:
I'm not sure it's correct detect as a gap the case this is the first fetch and `lastSeenOffset` is a special negative value (kafkaOffsetStart or kafkaOffsetEnd). I suggest:

```
if (lastSeenOffset < 0) {
  // If this is the first fetched records batch and we started reading from a special offset (e.g. kafkaOffsetStart or kafkaOffsetEnd)
  // then we don't want to report it as a gap.
  lastSeenOffset = <the offset of the first record>
}
```

Reply:
If you agree with this change, then please update the unit test too.

Reply:
because we ignore negative gaps, this works without the change. I updated the test